lib/src/config.cc

```diff
@@ -29,13 +30,27 @@ using namespace std;
 
 namespace hocon {
 
-    shared_config config::parse_file_any_syntax(std::string file_basename, config_parse_options options) {
+    shared_config config::parse_file_any_syntax(std::string file_basename, config_parse_options options, shared_full_current fpath) {
+        std::string file_dir(""), file_name("");
+        if (nullptr == fpath) {
+            fpath = make_shared<full_path_operator>();
+        }
+        extract_filename_from_path(file_basename, &file_dir, &file_name);
```

Comment:
My instinct is that this isn't the place we should be fixing this, but it'd take some work for me to follow up on.

From a quick glance at https://github.com/lightbend/config/blob/7b9e1539dd46385e9b83be43c4112a448129ac2f/config/src/main/java/com/typesafe/config/impl/Parseable.java#L639-L655, I think the mistake may be in not overriding `relative_to` on `parseable_file` (and possibly others) in https://github.com/puppetlabs/cpp-hocon/blob/master/lib/src/parseable.cc.

Reply:
> I left some pretty high-level comments. Do you have an interest in trying to improve this, or should we consider this an open problem for someone to build on your solution?

Thanks for code-reviewing and comments. I think we could left the problem open before solved. And I'll keep trying to figure another way out to fix it. Thanks again~ 
It's 3 in the morning and I need some sleep. Good day!